reporter/symbol_querier.go

```diff
@@ -31,30 +33,48 @@ type SymbolsQueryRequest struct {
 	Arch     string   `json:"arch" jsonapi:"attribute" validate:"required"`
 }
 
-type DatadogSymbolQuerier struct {
+type DatadogSymbolQuerier interface {
+	Start(ctx context.Context)
+	QuerySymbols(ctx context.Context, buildIDs []string, arch string) ([]SymbolFile, error)
+	ResetCallCount() int
+}
+
+type datadogSymbolQuerier struct {
```

Comment:
Nit: Maybe call this `simpleSymbolQuerier` ? Also any reason this is not exporter but `BatchedSymbolQuerier` is ? I think both could be unexported, no strong opinion.